workbench-core/datasets/src/dataSetService.ts

```diff
@@ -718,52 +743,85 @@ export class DataSetService {
       );
     }
 
-    const readOnly = permissionsData.permissions.some(({ accessLevel }) => accessLevel === 'read-only');
-
-    if (_.find(targetDS.externalEndpoints, (ep) => ep === externalEndpointName)) {
-      throw new EndPointExistsError(`'${externalEndpointName}' already exists in '${dataSetId}'.`);
+    if (targetDS.externalEndpoints?.find((ep) => ep === externalEndpointName)) {
```

Comment:
this needs to be reworked to use externalEndpointId because that is what is stored. Not a great idea to change that this late in the game.

Reply:
I'll remove it since it currently doesnt do anything.